.yarn/sdks/typescript/lib/tsserver.js

```diff
@@ -10,11 +10,41 @@ const absPnpApiPath = resolve(__dirname, relPnpApiPath);
 const absRequire = (createRequire || createRequireFromPath)(absPnpApiPath);
 
 const moduleWrapper = tsserver => {
+  const {isAbsolute} = require(`path`);
+  const {resolveVirtual} = require(`pnpapi`);
+
   // VSCode sends the zip paths to TS using the "zip://" prefix, that TS
   // doesn't understand. This layer makes sure to remove the protocol
   // before forwarding it to TS, and to add it back on all returned paths.
 
-  const {isAbsolute} = require(`path`);
+  // We also take the opportunity to turn virtual paths into physical ones;
+  // this makes is much easier to work with workspaces that list peer
+  // dependencies, since otherwise Ctrl+Click would bring us to the virtual
+  // file instances instead of the real ones.
+
+  function toVSCodePath(str) {
```

Comment:
It's not only for VSCode though, the other editors also pass through this :thinking: